hs/src/Reach/Eval/Core.hs

```diff
@@ -5209,7 +5209,7 @@ doToConsensus ks (ToConsensusRec {..}) = locAt slptc_at $ do
   let tc_send = fmap snd tc_send'
   let msg_dass = fmap ds_msg tc_send
   let msg_dass_t = transpose $ M.elems $ msg_dass
-  let get_msg_t = typeEqs . map argTypeOf
+  let get_msg_t = typeEqs . map (\x -> (Nothing, argTypeOf x))
   msg_ts <- mapM get_msg_t msg_dass_t
```

Comment:
`tc_send` is the ToConsensus send structure, which is an instance of a `DLSend` --- https://github.com/reach-sh/reach-lang/blob/master/hs/src/Reach/AST/DLBase.hs#L1390 --- that structure represents their piece of the `publish` (constructed on line 5204)

The correct source location for this `x`'s type is in the result of the `evalId` on line 5201.

Basically, we want to change
```
                ds_msg <- mapM (\v -> snd <$> (compileTypeOf =<< ensure_public =<< evalId "publish msg" v)) msg
```
into
```
                (ds_msg_at, ds_msg) <- unzip <$> mapM (\v -> do
                  sv <- evalId "publish msg" v
                  va <- snd <$> (compileTypeOf =<< ensure_public =<< sv)
                  return (srcLocOf sv, va)) msg
```
(Or make it so that the type of `ds_msg` is `[(SrcLoc, DLArg)]` rather than `[DLArg]`.